internal/juju/deployments.go

```diff
@@ -0,0 +1,178 @@
+package juju
+
+import (
+	"errors"
+	"fmt"
+	"github.com/juju/charm/v8"
+	jujuerrors "github.com/juju/errors"
+	"github.com/juju/juju/api/client/application"
+	apiapplication "github.com/juju/juju/api/client/application"
+	apicharms "github.com/juju/juju/api/client/charms"
+	apiclient "github.com/juju/juju/api/client/client"
+	apimodelconfig "github.com/juju/juju/api/client/modelconfig"
+	"github.com/juju/juju/cmd/juju/application/utils"
+	"github.com/juju/juju/core/constraints"
+	"github.com/juju/juju/environs/config"
+	"github.com/juju/juju/version"
+	"github.com/juju/names/v4"
+)
+
+type deploymentsClient struct {
+	ConnectionFactory
+}
+
+type CreateDeploymentInput struct {
+	ApplicationName string
+	ModelUUID       string
+	CharmName       string
+	CharmChannel    string
+	CharmSeries     string
+	CharmRevision   int
+}
+
+func (c deploymentsClient) CreateDeployment(input *CreateDeploymentInput) (string, error) {
```

Comment:
In the models.go file we've just used Create, Read, etc. Should we keep this consistent, or should the others be updated to be more explicit (CreateModel etc.)